src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,671 @@
+	 *
+	 * It adds or removes CSS classes in the current HTML element based on the
+	 * evaluation of its associated references.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Interactivity_API_Directives_Processor $p               The directives processor instance.
+	 * @param array                                     $context_stack   The reference to the context stack.
+	 * @param array                                     $namespace_stack The reference to the store namespace stack.
+	 */
+	private function data_wp_class_processor( WP_Interactivity_API_Directives_Processor $p, array &$context_stack, array &$namespace_stack ) {
+		if ( ! $p->is_tag_closer() ) {
+			$all_class_directives = $p->get_attribute_names_with_prefix( 'data-wp-class--' );
+
+			foreach ( $all_class_directives as $attribute_name ) {
+				list( , $class_name ) = $this->extract_prefix_and_suffix( $attribute_name );
+				if ( empty( $class_name ) ) {
+					return;
+				}
+
+				$attribute_value = $p->get_attribute( $attribute_name );
+				$result          = $this->evaluate( $attribute_value, end( $namespace_stack ), end( $context_stack ) );
+
+				if ( $result ) {
+					$p->add_class( $class_name );
+				} else {
+					$p->remove_class( $class_name );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Processes the `data-wp-style` directive.
+	 *
+	 * It updates the style attribute value of the current HTML element based on
+	 * the evaluation of its associated references.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Interactivity_API_Directives_Processor $p               The directives processor instance.
+	 * @param array                                     $context_stack   The reference to the context stack.
+	 * @param array                                     $namespace_stack The reference to the store namespace stack.
+	 */
+	private function data_wp_style_processor( WP_Interactivity_API_Directives_Processor $p, array &$context_stack, array &$namespace_stack ) {
+		if ( ! $p->is_tag_closer() ) {
+			$all_style_attributes = $p->get_attribute_names_with_prefix( 'data-wp-style--' );
+
+			foreach ( $all_style_attributes as $attribute_name ) {
+				list( , $style_property ) = $this->extract_prefix_and_suffix( $attribute_name );
+				if ( empty( $style_property ) ) {
+					continue;
+				}
+
+				$directive_attribute_value = $p->get_attribute( $attribute_name );
+				$style_property_value      = $this->evaluate( $directive_attribute_value, end( $namespace_stack ), end( $context_stack ) );
+				$style_attribute_value     = $p->get_attribute( 'style' );
+				$style_attribute_value     = ( $style_attribute_value && ! is_bool( $style_attribute_value ) ) ? $style_attribute_value : '';
+
+				/*
+				 * Checks first if the style property is not falsy and the style
+				 * attribute value is not empty because if it is, it doesn't need to
+				 * update the attribute value.
+				 */
+				if ( $style_property_value || ( ! $style_property_value && $style_attribute_value ) ) {
```

Comment:
This can be simplified to remove `! $style_property_value` because it will always evaluate to truthy here.

```suggestion
				if ( $style_property_value || $style_attribute_value ) {
```

Reply:
True. Good catch. Done in https://github.com/WordPress/wordpress-develop/pull/5953/commits/61d896fe5dba0a6a8dfa9bbc902702173d138aaf.